crates/wasmtime/src/runtime/vm/libcalls.rs

```diff
@@ -655,6 +657,60 @@ fn update_mem_size(instance: &mut Instance, num_pages: u32) {
     }
 }
 
+fn trap(_instance: &mut Instance, code: u32) -> Result<(), TrapReason> {
+    Err(TrapReason::Wasm(
+        wasmtime_environ::Trap::from_u8(code.try_into().unwrap()).unwrap(),
+    ))
+}
```

Comment:
This seems to be identical to the component trapping libcall -- is there a reason we need two copies and can't just use this one everywhere?

Reply:
There's some subtle differences in the ABI like the exact return value and the first parameter between the component/core wasm side of things. The component side of things is a bit unfortunate because it's only needed for a degenerate case of always-trapping adapters, but I'll probably leave the minor duplication for now in the hopes this can be refactored away in the future

Reply:
Ah gotcha. SGTM